__tests__/useSelectedItems.test.tsx

```diff
@@ -0,0 +1,155 @@
+import { renderHook, act } from "@testing-library/react-hooks";
+
+import useSelectedItems from "../src/index";
+
+const initialExampleItems = [
+  { id: 1, text: "What's up" },
+  { id: 2, text: "Hey there" },
+  { id: 3, text: "How you doin'" },
+];
+
+interface ExampleItem {
+  id: number;
+  text: string;
+}
+
+describe("useSelectedItems", () => {
+  it("should return all items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    const expectedItems = initialExampleItems.map((item) => ({
+      ...item,
+      isSelected: false,
+    }));
+
+    expect(result.current.items).toEqual(expectedItems);
+  });
+
+  it("should render with initialSelectedItems", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialSelectedItems: [initialExampleItems[0]],
+      initialItems: initialExampleItems,
+    }));
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    expect(result.current.items).toContainEqual(
+      expect.objectContaining({ ...initialExampleItems[0], isSelected: true }),
+    );
+  });
+
+  it("should render with empty items", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+    }));
+
+    expect(result.current.items).toEqual([]);
+  });
+
+  it("should toggle a single item", () => {
+    const { result } = renderHook(() => useSelectedItems<ExampleItem, ExampleItem["text"]>({
+      itemIdentifierKey: "id",
+      initialItems: initialExampleItems,
+    }));
+
+    act(() => {
+      result.current.toggleSingleItem(1);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+
+    act(() => {
+      result.current.toggleSingleItem(2);
+    });
+
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[0]),
+    );
+    expect(result.current.selectedItems).toContainEqual(
+      expect.objectContaining(initialExampleItems[1]),
+    );
```

Comment:
I would suggest to "merge" these assertions since the two items can be verified at the same time.

```expect(result.current.selectedItems).toEqual([initialExampleItems[0], initialExampleItems[1]]);```

But this it's not going to work if the current implementation since I also noticed an error in the `selectedItems` implementation. It should't have the `isSelected` state being returned within the items since it's already selected... kind of redundant. 

In order to fix it, the selected items should be returned like the following:
```
const selectedItems = useMemo(() => (
    items
      .filter(item => item.isSelected)
      .map(({ isSelected: _isSelected, ...rest }) => ({
        ...rest,
      })) as T[]
  ), [items]);
  ```
  
I will make this change in another PR and then you can pull the changes here

Reply:
Changes to `selectedItems` applied at 4b52965826caf784ad47758f8c2e4d1edf549bb5

Now you can pull the changes from `master`